SignalR clarify connectionID access options (#28343)
dotnet/AspNetCore.Docs: aspnetcore/signalr/hubcontext.md

> [!NOTE]
> When client methods are called from outside of the `Hub` class, there's no caller associated with the invocation. Therefore, there's no access to the `ConnectionId`, `Caller`, and `Others` properties.
>
> If your application needs to map a user to the connection ID and persist that mapping, you can do one of the following:
>
> - Persist mapping of single or multiple connections as groups. See [Groups in SignalR](xref:signalr/groups#groups-in-signalr) for more information.
> - Retain connection and user information in a singleton service. See [Inject services into a hub](xref:signalr/hubs#inject-services-into-a-hub) for more information.

> > - Retain connection and user information in a dictionary that is stored in memory.

Reviewer: It's not just in-memory, you can store the info in a database or any other storage method.

> [!NOTE]
> When client methods are called from outside of the `Hub` class, there's no caller associated with the invocation. Therefore, there's no access to the `ConnectionId`, `Caller`, and `Others` properties.
>

> > If your application needs to map a user to the connection ID and persist that mapping, you can do one of the following:

Reviewer 2: ```suggestion
> Apps that need to map a user to the connection ID and persist that mapping, can do one of the following:
```

Reviewer 2: [Use bigger ideas, fewer words](https://learn.microsoft.com/en-us/style-guide/top-10-tips-style-voice#use-bigger-ideas-fewer-words) : Our modern design hinges on crisp minimalism. Shorter is always better.
[Short sentences and fragments are easier to scan and read.](https://learn.microsoft.com/en-us/style-guide/brand-voice-above-all-simple-human#a-focus-on-the-customer) Prune every excess word.

What about consultants who are working on an app? This repo uses a more formal style.

Author: I added a note to also fix those in the original published doc that text snippet was derived from, which I will have a chance to rewrite.  Thanks.

> [!NOTE]
> When client methods are called from outside of the `Hub` class, there's no caller associated with the invocation. Therefore, there's no access to the `ConnectionId`, `Caller`, and `Others` properties.
>
> If your application needs to map a user to the connection id and persist that mapping, you can use one of the following:
>
> - [The User ID Provider](#IUserIdProvider)
> - [In-memory storage](#inmemory), such as a dictionary
> - [SignalR group for each user](#groups)
> - [Permanent, external storage](#database), such as a database table or Azure table storage
>

> > For more information see, [Mapping SignalR Users to Connections](xref:signalr/overview/guide-to-the-api/mapping-users-to-connections).

Reviewer: Wrong docs. That's for ASP.NET SignalR

Author: Thanks, pulling back this draft.

> >   - In-memory storage in a dictionary.

Author: I have moved the storing options to under the singleton service. I wasn't catching on that the storage options were referring only to the singleton service.  Thanks!
